app/views/Project/Tagging/Sources/SourcesFilter/index.tsx

```diff
@@ -24,62 +26,284 @@ import {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        sourcePriorityOptions: __type(name: "LeadPriorityEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        sourceConfidentialityOptions: __type(name: "LeadConfidentialityEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        project(id: $projectId) {
+            members {
+                id
+                displayName
+                firstName
+                lastName
+            }
+            analysisFramework {
+                filters {
+                    filterType
+                    key
+                    properties
+                    title
+                    widgetType
+                }
+            }
+        }
+        organizationTypes {
+            results {
+                id
+                title
+            }
+        }
+        emmEntititiesOptions: __type(name: "EmmEntityType") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        emmRiskFactorsOptions: __type(name: "EmmKeyRiskFactorType") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        emmKeywordsOptions: __type(name: "EmmKeyWordType") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        entryTypeOptions: __type(name: "EntryTagTypeEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        commentStatusOptions: __type(name: "EntryFilterCommentStatusEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+    }
+`;
+
+const userLabelSelector = (
+    d: NonNullable<SourceFilterOptionsQuery['project']>['members'][number],
+) => d.displayName ?? `${d.firstName} ${d.lastName}`;
+const userKeySelector = (
+    d: NonNullable<SourceFilterOptionsQuery['project']>['members'][number],
+) => d.id;
+const organizationTypeKeySelector = (
+    d: NonNullable<NonNullable<NonNullable<SourceFilterOptionsQuery['organizationTypes']>>['results']>[number],
+) => d.id;
+const organizationTypeLabelSelector = (
+    d: NonNullable<NonNullable<NonNullable<SourceFilterOptionsQuery['organizationTypes']>>['results']>[number],
+) => d.title;
+
+function getProjectSourcesQueryVariables(
+    filters: SourcesFilterFields,
+) {
+    const {
+        createdAt: createdAtRaw,
+        publishedOn: publishedOnRaw,
+        entriesFilterData,
+        ...leadsFilters
+    } = filters;
+
+    const createdAt = getValidDateRangeValues(createdAtRaw);
+    const publishedOn = getValidDateRangeValues(publishedOnRaw);
```

Comment:
We should not transform publishedOn.
@thenav56 @AdityaKhatri we are storing publishedOn as date right?

Reply:
Yes published on is a date field.